sklearn/preprocessing/callable_transformer.py

```diff
@@ -0,0 +1,44 @@
+from ..base import BaseEstimator, TransformerMixin
+from ..utils import check_array
+
+
+class CallableTransformer(BaseEstimator, TransformerMixin):
+    """Allows the construction of a transformer from an arbitrary callable.
+
+    Parameters
+    ----------
+    func : callable, optional default=None
+        The callable to use for the transformation. This will be passed
+        the same arguments as transform, with args and kwargs forwarded.
+        If func is None, then func will be the identity function.
+    validate : bool, optional default=True
+        Indicate that the input X array should be checked before calling
+        func. If validate is false, there will be no input validation.
+    accept_sparse : boolean, optional
+        Indicate that func accepts a sparse matrix as input.
+    args : tuple, optional
+        A tuple of positional arguments to be passed to func. These will
+        be passed after X and y.
+    kwargs : dict, optional
+        A dictionary of keyword arguments to be passed to func.
+
```

Comment:
Suggestion for example: `func=partial(getattr, 'data')`, and feed the transformer a dict `{'data': X, other stuff...}`
Sorry, I can't think of better names.

I often end up doing this when I do e.g. text classification on conversational data and I have messages in both directions. I store my samples as {'from': from, 'to': to} and use a `FeatureUnion` of two pipelines, each grabbing the respective field and then doing a CountVectorizer.